internal/util/paths/paths.go

```diff
@@ -79,17 +78,24 @@ func WriteString(path string, content string) error {
 	return err
 }
 
-// Expands leading tilde.
-func ExpandTilde(path string) (string, error) {
+// Expands home directory to absolute path.
+// If no expansion is required or possible, the string is returned unaltered.
+// If there is an error, an empty string is returned.
+func ExpandHomeDir(path string) (string, error) {
+
+	// Expand in case there are environment variables on the path
+	path = os.ExpandEnv(path)
 	usr, err := user.Current()
 	if err != nil {
-		return "", fmt.Errorf("failed to determine current user")
+		return "", err
 	}
 	home := usr.HomeDir
```

Comment:
> HomeDir is the path to the user's home directory (_if they have one_).

https://pkg.go.dev/os/user#User emphasis mine

If a user doesn't have a home, but we need one (`~` is specified), we should create an error. We shouldn't be using an empty string as `home`

Reply:
I second that.